zipkin/getmetrics_zipkin.py

```diff
@@ -0,0 +1,524 @@
+    return UNDERSCORE.sub(".", string)
+
+
+def make_safe_metric_key(string):
+    """ make safe string already handles this"""
+    return make_safe_string(string)
+
+
+def make_safe_string(string):
+    """
+    Take a single string and return the same string with spaces, slashes,
+    underscores, and non-alphanumeric characters subbed out.
+    """
+    string = SPACES.sub("-", string)
+    string = SLASHES.sub(".", string)
+    string = UNDERSCORE.sub(".", string)
+    string = NON_ALNUM.sub("", string)
+    return string
+
+
+def set_logger_config(level):
+    """Set up logging according to the defined log level"""
+    # Get the root logger
+    logger_obj = logging.getLogger(__name__)
+    # Have to set the root logger level, it defaults to logging.WARNING
+    logger_obj.setLevel(level)
+    # route INFO and DEBUG logging to stdout from stderr
+    logging_handler_out = logging.StreamHandler(sys.stdout)
+    logging_handler_out.setLevel(logging.DEBUG)
+    # create a logging format
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(process)d - %(threadName)s - %(levelname)s - %(message)s')
+    logging_handler_out.setFormatter(formatter)
+    logger_obj.addHandler(logging_handler_out)
+
+    logging_handler_err = logging.StreamHandler(sys.stderr)
+    logging_handler_err.setLevel(logging.WARNING)
+    logger_obj.addHandler(logging_handler_err)
+    return logger_obj
+
+
+def print_summary_info():
+    # info to be sent to IF
+    post_data_block = '\nData sent to IF:'
+    post_data = initialize_api_post_data()
+    for i in post_data.keys():
+        post_data_block += '\n\t' + i + ': ' + str(post_data[i])
+    logger.debug(post_data_block)
+
+    # variables from agent-specific config
+    agent_data_block = '\nAgent settings:'
+    for j in agent_config_vars.keys():
+        agent_data_block += '\n\t' + j + ': ' + str(agent_config_vars[j])
+    logger.debug(agent_data_block)
+
+
+def initialize_data_gathering():
+    reset_track()
+    track['chunk_count'] = 0
+    track['entry_count'] = 0
+
+    start_data_processing()
+
+    # last chunk
+    if len(track['current_row']) > 0 or len(track['current_dict']) > 0:
+        send_data_wrapper()
+
+    logger.debug("Total chunks created: " + str(track['chunk_count']))
+    logger.debug("Total " + track['mode'].lower() + " entries: " + str(track['entry_count']))
+
+
+def reset_track():
+    """ reset the track global for the next chunk """
+    track['start_time'] = time.time()
+    track['line_count'] = 0
+    track['current_row'] = []
+    track['current_dict'] = dict()
+
+
+###################################
+# Functions to handle Metric data #
+###################################
+def metric_handoff(timestamp, field_name, instance, data):
+    append_metric_data_to_entry(timestamp, field_name, instance, data)
+    track['line_count'] += 1
+    track['entry_count'] += 1
+    if len(track['current_dict']) >= if_config_vars['chunkLines']:
```

Comment:
please convert if_config_vars['chunkLines'] to int

Reply:
now converted to int when grabbed from the config file